compiler/src/iree/compiler/Codegen/LLVMGPU/Passes.cpp

```diff
@@ -560,6 +560,11 @@ void addGPUTransformDialectPasses(OpPassManager &passManager) {
   //      schedule once applied.
   //   2. if transform.do_not_dce_operands ops are introduced.
   passManager.addPass(createDropSchedulePass());
+
+  // TODO: Remove the following pass the plumb support for #hal.descriptor_type
+  // memory space through the stack.
+  passManager.nest<ModuleOp>().addNestedPass<func::FuncOp>(
```

Comment:
If this is needed on all passes we could just add this to `addLowerToLLVMGPUPasses`?

Reply:
That would be fine with me too; as far as I can tell if there are any residual descriptors it will always fail to lower to a valid memory space but would be good to confirm with someone more familiar with the different CUDA passes that it is fine to always run this.

Reply:
@antiagainst could you PTAL.

Reply:
Yeah we should try to put such cleanup passes at late as possible in the pipeline so that we can have the initial pipelines look more similar (and sharable) between LLVMGPU and SPIR-V. I cannot recall why I put it as-is; probably due to some other issues when initially plumbing this through.

Quinn could you try to move it to `addLowerToLLVMGPUPasses`?

Reply:
Sounds good!